cost/azure/old_snapshots/azure_delete_old_snapshots.pt

```diff
@@ -0,0 +1,237 @@
+  iterate $ds_subscriptions
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host "management.azure.com"
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"), "/providers/Microsoft.Compute/snapshots"])
+    query "api-version","2019-07-01"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item,"id")
+      field "name", jmes_path(col_item,"name")
+      field "location", jmes_path(col_item, "location")
+      field "tags", jmes_path(col_item,"tags")
+      field "timeCreated", jmes_path(col_item,"properties.timeCreated")
+      field "subscriptionId",val(iter_item,"subscriptionId")
+      field "subscriptionName",val(iter_item,"displayName")
+    end
+  end
+end
+
+datasource "ds_filter_snapshots" do
+  run_script $js_filter_snapshots, $param_exclusion_tag_key, $param_snapshot_age, $ds_azure_get_snapshots
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_filter_snapshots", type: "javascript" do
+  parameters "param_exclusion_tag_key", "param_snapshot_age", "ds_azure_get_snapshots"
+  result "res"
+  code <<-EOS
+   var res = [];
+   _.each(ds_azure_get_snapshots, function(snapshot){
+      var tags='';
+      var expectedDate=new Date();
+	  var is_tag_matched=false;
+	  var date1 = new Date(snapshot['timeCreated']);
+      var date2 = new Date(); 
+      var Difference_In_Time = date2.getTime() - date1.getTime(); 
+      var age = parseInt(Difference_In_Time / (1000 * 3600 * 24));
+      var createdTime=new Date(snapshot['timeCreated']);
+      expectedDate.setDate(expectedDate.getDate()- parseInt(param_snapshot_age));	  
+	  for(var k=0;k<param_exclusion_tag_key.length;k++){
+	     if (_.has(snapshot.tags, param_exclusion_tag_key[k])) {
+            is_tag_matched=true
+      }
+	  }	  
+      if (!is_tag_matched) {
+         if(expectedDate.getTime()<createdTime.getTime()){
+            res.push({
+              snapshot_id:snapshot['id'],
+              name:snapshot['name'],
+              location:snapshot['location'],
+              timeCreated:snapshot['timeCreated'],
+              subscriptionId:snapshot['subscriptionId'],
+              subscriptionName:snapshot['subscriptionName'],
+			  age:age
+            })
+        }
+      }    
+   })
+   res = _.sortBy(res, 'subscriptionName');
+   res = _.sortBy(res, 'location');
+   res = _.sortBy(res, 'age');
+  EOS
+end
+
+policy "pol_azure_delete_old_snapshots" do
+  validate $ds_filter_snapshots do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Old Snapshots Found"
+    detail_template <<-EOS
+The following {{ len data }} delete old snapshots, for Account: {{ rs_project_name }} (ID: {{ rs_project_id }}), have exceeded the specified age of: {{ parameters.param_unattached_days }} days old\n
```

Comment:
This sentence doesn't read well.   consider removing **delete**

Reply:
Updated the sentence.